src/components/Books.js

```diff
@@ -1,31 +1,35 @@
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
 import Book from './Book';
 import AddNewBook from './AddNewBook';
+import { fetchBooks } from '../redux/books/booksSlice';
 
 const Books = () => {
   const { books } = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch, books]);
```

Comment:
- [ ] When I go to the main page the fetchBooks function is called endlessly because you added both dispatch and books inside the dependency array of `useEffect`, Kindly place `dispatch` only in the dependency array or leave it empty so the fetchBooks function will be called only once.